sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java

```diff
@@ -0,0 +1,1014 @@
+            new TextDocumentInput("0", "The hotel was dark and unclean. The restaurant had amazing gnocchi."),
+            new TextDocumentInput("1", "The restaurant had amazing gnocchi. The hotel was dark and unclean.")
+        );
+        testRunner.accept(textDocumentInputs, setTextAnalyticsRequestOptions());
+    }
+
+    static List<String> getSentimentInput() {
+        return Arrays.asList("The hotel was dark and unclean. The restaurant had amazing gnocchi.",
+            "The restaurant had amazing gnocchi. The hotel was dark and unclean.");
+    }
+
+    String getEndpoint() {
+        return interceptorManager.isPlaybackMode()
+            ? "http://localhost:8080"
+            : Configuration.getGlobalConfiguration().get("AZURE_TEXT_ANALYTICS_ENDPOINT");
+    }
+
+    /**
+     * Helper method to verify batch result.
+     *
+     * @param actualResult DocumentResultCollection<> returned by the API.
+     * @param testApi the API to test.
+     */
+    <T> void validateBatchResult(DocumentResultCollection<T> actualResult,
+        DocumentResultCollection<T> expectedResult, TestEndpoint testApi) {
+        // assert batch result
+        assertEquals(expectedResult.getModelVersion(), actualResult.getModelVersion());
+        if (this.showStatistics) {
+            validateBatchStatistics(expectedResult.getStatistics(), actualResult.getStatistics());
+        }
+        validateDocuments(expectedResult, actualResult, testApi);
+    }
+
+    /**
+     * Helper method to verify documents returned in a batch request.
+     *
+     * @param expectedResult the expected result collection..
+     * @param actualResult the actual result collection returned by the API.
+     * @param testApi the API to test.
+     */
+    private <T> void validateDocuments(DocumentResultCollection<T> expectedResult,
+        DocumentResultCollection<T> actualResult, TestEndpoint testApi) {
+        switch (testApi) {
+            case LANGUAGE:
+                final List<DetectLanguageResult> detectLanguageResults = expectedResult.stream()
+                    .filter(element -> element instanceof DetectLanguageResult)
+                    .map(element -> (DetectLanguageResult) element)
+                    .collect(Collectors.toList());
+
+                final List<DetectLanguageResult> actualDetectLanguageResults = actualResult.stream()
+                    .filter(element -> element instanceof DetectLanguageResult)
+                    .map(element -> (DetectLanguageResult) element)
+                    .collect(Collectors.toList());
+                assertEquals(detectLanguageResults.size(), actualDetectLanguageResults.size());
+
+                actualDetectLanguageResults.forEach(actualItem -> {
+                    Optional<DetectLanguageResult> optionalExpectedItem = detectLanguageResults.stream().filter(
+                        expectedEachItem -> actualItem.getId().equals(expectedEachItem.getId())).findFirst();
```

Comment:
find first? what if there are multiple items in the set with the same id? I'd expect it to still fail.

Reply:
updated to fail when size returned more than 1